bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -1174,24 +1208,19 @@ private File findFile(long logId) throws FileNotFoundException {
      * @throws IOException
      */
     public void scanEntryLog(long entryLogId, EntryLogScanner scanner) throws IOException {
-        // Buffer where to read the entrySize (4 bytes) and the ledgerId (8 bytes)
-        ByteBuf headerBuffer = Unpooled.buffer(4 + 8);
-        BufferedReadChannel bc;
-        // Get the BufferedChannel for the current entry log file
+        EntryLogBufferedReadChannel bc = null;
```

Comment:
There's a lot of code moved around here for what seems to be just exception. Couldn't you have just released the bc object in the final finally block.

Reply:
I just move previous exception handling code's location. I think keeping original's exception handling logic is necessary.

Reply:
I'm questioning it, because it creates noise the in patch. Theres a bunch of lines added and deleted, which are likely exactly the same, but they need to be checked anyhow. If you need to refactor, it's better to do in a separate patch which doesn't change any logic.